app.js

```diff
@@ -165,88 +165,111 @@ app.get("/login", routes.login);
 // resource proxying for http-on-https
 webmakerProxy(app, middleware.checkForAuth);
 
-// Entry point for authenticated users
+/*
+ * Main routes for thimble.webmaker.org
+ */
+// Entry point for all users
 app.get('/',
+        middleware.setUserIfTokenExists,
+        middleware.setPublishUser,
+        routes.main.root);
+
+// Main route for authenticated users
+app.get('/user/:username/:projectId',
```

Comment:
This was purely for semantics. I can't do `/:projectId` because no other route would work. I wanted to do `/projects/:projectId` but I couldn't because there already is a route like that (new projects), and I didn't want to do `/user/:projectId` because I don't think it would make sense because it would be a mismatch between class and identifier (user->projectId). So just to prettify the url a bit (what the user will see), I added the username.